src/tool_parsecfg.c

```diff
@@ -86,14 +99,20 @@ int parseconfig(const char *filename, struct GlobalConfig *global)
               *lastdirchar = 0;
               /* If we have enough space, build the RC filename */
               remaining = sizeof(filebuffer) - strlen(filebuffer);
-              if(strlen(CURLRC) < remaining - 1) {
+              if(strlen(filename) < remaining - 1) {
                 msnprintf(lastdirchar, remaining,
-                          "%s%s", DIR_CHAR, CURLRC);
-                /* Don't bother checking if it exists - we do that later */
-                filename = filebuffer;
+                          "%s%s", DIR_CHAR, filename);
+                file = fopen(filebuffer, FOPEN_READTEXT);
+                if(file != NULL) {
+                  filename = filebuffer;
+                  break;
+                }
               }
             }
           }
+
+          /* If we didn't find CURLRC in either location we loop round and try
+             CURLRC_LEGACY next */
```

Comment:
I have to say that this indentation change makes the patch quite hard to review, at least for me.

Besides, I do not like that `CURLRC_LEGACY` is defined but then used in anything but an `#ifdef CURLRC_LEGACY` code block.

Therefore, I would like to suggest a different strategy, based on a simple and elegant `goto`. In particular:

- add a label, say, `retry_reading_curlrc:` instead of the `for` statement,
- Replace the comment `/* Don't bother checking if it exists - we do that later */` and the line following it by something like this:

```c
file = fopen(filebuffer, FOPEN_READTEXT);
if(file != NULL) {
  filename = filebuffer;
}
#ifdef CURLRC_LEGACY
else if (strcmp(CURLRC_LEGACY, filename)) {
    filename = CURLRC_LEGACY;
    goto retry_reading_curlrc;
}
#endif
```

That will be much easier to review, and therefore it will be much harder for bugs to creep in.